libvast/src/format/mrt.cpp

```diff
@@ -14,47 +14,81 @@ namespace mrt {
 
 namespace table_dump_v2 {
 
-const auto peer_entries_type = record_type{{
-  {"type", count_type{}},
-  {"bgp_id", count_type{}},
-  {"ip_address", address_type{}},
-  {"as", count_type{}},
-}};
+  const auto peer_entry_type = record_type{{
+    {"index", count_type{}},
+    {"bgp_id", count_type{}},
+    {"ip_address", address_type{}},
+    {"as", count_type{}},
+  }}.name("mrt::table_dump_v2::peer_entry");
 
-const auto peer_index_table_type = record_type{{
-  {"collector_bgp_id", count_type{}},
-  {"view_name", string_type{}},
-  {"ip_address", address_type{}},
-  {"peer_entries", vector_type{peer_entries_type}}
-}};
+  const auto rib_entry_type = record_type{{
+    {"peer_index", count_type{}},
+    {"prefix", subnet_type{}},
+    {"as_path", vector_type{count_type{}}},
+    {"origin_as", count_type{}},
+    {"origin", string_type{}.attributes({{"skip"}})},
+    {"nexthop", address_type{}},
+    {"local_pref", count_type{}},
+    {"med", count_type{}},
+    {"community", vector_type{count_type{}}},
+    {"atomic_aggregate", boolean_type{}},
+    {"aggregator_as", count_type{}},
+    {"aggregator_ip", address_type{}},
+  }}.name("mrt::table_dump_v2::rib_entry");
 
 } // namespace table_dump_v2
 
-
 namespace bgp4mp {
 
-// TODO: Define this type.
-const auto message_as4_type = type{};
+// TODO: Define this types.
 
 } // namespace bgp4mp
 
 namespace {
 
 struct factory {
-  value operator()(table_dump_v2::peer_index_table& /* x */) const {
-    // TODO: Implement this function.
-    return {nil, table_dump_v2::peer_index_table_type};
+  factory(std::queue<event>& events, uint32_t timestamp) : events_(events) {
+    timestamp_ = timestamp;
+  }
+
+  void addTimestamp(event& e) {
+    std::chrono::duration<uint32_t> since_epoch{timestamp_};
+    auto ts = timestamp{std::chrono::duration_cast<timespan>(since_epoch)};
+    e.timestamp(ts);
+  }
```

Comment:
You can get rid of this function by performing this conversion in the constructor.